src/python/pants/engine/target.py

```diff
@@ -1705,12 +1720,25 @@ def compute_value(
 class StringSequenceField(SequenceField[str]):
     expected_element_type = str
     expected_type_description = "an iterable of strings (e.g. a list of strings)"
+    valid_choices: ClassVar[Optional[Union[Type[Enum], Tuple[str, ...]]]] = None
 
     @classmethod
     def compute_value(
         cls, raw_value: Optional[Iterable[str]], address: Address
     ) -> Optional[Tuple[str, ...]]:
-        return super().compute_value(raw_value, address)
+        value_or_default = super().compute_value(raw_value, address)
+        if value_or_default and cls.valid_choices is not None:
```

Comment:
I think this should be `if value_or_default is not None and cls.valid_choices is not None` ?

Otherwise an empty string would bypass the validity check, but may not be valid.

Reply:
I was trying to short circuit the logic. If `value_or_default` is an empty list, then there's nothing to validate. Also, if it were an empty string, the logic below still wouldn't catch it.

Reply:
type checking happens in `super().compute_value(raw_value, address)`, which is implemented in `SequenceField` and uses `ensure_list()` to make sure all elements are of the expected type.

So, we could check for `if value_or_default is not None` but then we'll needlessly calculate `valid_choices` for the empty sequence case. Do you want to do that?

Reply:
Oh right, sorry, this is always a list.